compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -993,20 +993,22 @@ object Parsers {
       else {
         val saved = placeholderParams
         placeholderParams = Nil
+
+        def wrapPlaceholders(t: Tree) = try
+          if (placeholderParams.isEmpty) t
+          else new WildcardFunction(placeholderParams.reverse, t)
+        finally placeholderParams = saved
+
         val t = expr1(location)
         if (in.token == ARROW) {
-          placeholderParams = saved
-          closureRest(start, location, convertToParams(t))
+          placeholderParams = Nil
```

Comment:
I don't understand why it's okay to forget the value of `placeholderParams` here, could you maybe add a comment clarifying this?